src/backend/services/logger/strategies/structured_log.py

```diff
@@ -31,28 +32,40 @@ def get_context_log(ctx: Context) -> dict:
 
 class StructuredLogging(BaseLogger):
     def __init__(self, level: str = "info"):
-        structlog.configure(
-            processors=[
-                structlog.processors.add_log_level,
-                structlog.processors.EventRenamer("msg"),
-                structlog.processors.TimeStamper(fmt="iso"),
-                structlog.processors.StackInfoRenderer(),
-                structlog.processors.format_exc_info,
-                structlog.processors.UnicodeDecoder(),
+        self.setup(level)
+
+        self.logger = structlog.get_logger()
+
+    def setup(self, level: str = "info"):
+        structlog.contextvars.clear_contextvars()
+
+        shared_processors = [
+            structlog.processors.add_log_level,
+            structlog.processors.EventRenamer("msg"),
+            structlog.processors.TimeStamper(fmt="iso"),
+            structlog.processors.StackInfoRenderer(),
+            structlog.processors.UnicodeDecoder(),
+        ]
+
+        # If running in a terminal, use colored output
+        # Otherwise, use JSON output
+        if sys.stderr.isatty():
```

Comment:
According to their official documentation, this works, but there's no way to test it until we deploy a new version.

It looks way better in the console:
<img width="987" alt="image" src="https://github.com/user-attachments/assets/b0eb29dc-4221-447e-9afb-d105c22b7797">